src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -0,0 +1,152 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.filters.WellformedReadFilter;
+import org.broadinstitute.hellbender.utils.iterators.IntervalOverlappingIterator;
+import org.broadinstitute.hellbender.utils.iterators.ReadFilteringIterator;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.downsampling.DownsamplingMethod;
+import org.broadinstitute.hellbender.utils.locusiterator.LocusIteratorByState;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A LocusWalker is a tool that processes reads that overlap a single position in a reference at a time from
+ * one or multiple sources of reads, with optional contextual information from a reference and/or sets of
+ * variants/Features.
+ *
+ * LocusWalker authors must implement the apply() method to process each position, and may optionally implement
+ * onTraversalStart() and/or onTraversalSuccess().
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class LocusWalker extends GATKTool {
+
+    @Argument(fullName = "disable_all_read_filters", shortName = "f", doc = "Disable all read filters", common = false, optional = true)
+    public boolean disableAllReadFilters = false;
+
+    /**
+     * Should the LIBS keep unique reads?
+     */
+    protected static boolean KEEP_UNIQUE_READ_LIST_IN_LIBS = false;
```

Comment:
I didn't make it final, because extended classes may need to keep the unique read list in the iterator and there should be a way of setting it. I could make it non-static and/or make a private getter for this behaviour instead of an static variable.